contrib/test_jenkins.sh

```diff
@@ -975,18 +975,19 @@ test_ucp_dlopen() {
 }
 
 test_ucm_hooks() {
+    total=30
     echo "==== Running UCM Bistro hook test ===="
-    for i in $(seq 1 30); do
-        echo -n "."
-        ./test/apps/test_hooks -n $((`nproc`*4/5+1)) >test_hooks.log 2>&1 || \
+    for i in $(seq 1 $total); do
+        echo "iteration $i/$total"
+        timeout 10 ./test/apps/test_hooks -n $((`nproc`*4/5+1)) >test_hooks.log 2>&1 || \
```

Comment:
why *4/5+1?

Reply:
take ~80% percent of the cores but at least one cpu.

Reply:
why not all the cores?

Reply:
I would even run more threads than cores (or random number of threads every time) to make more stress

Reply:
fixed

Reply:
out of curiosity, i checked by starting 522 threads on a 28 cores machine that we can repro issue without fix, added delay, issue still visible, then using the lock and delay, issue cannot be seen. (with 522 that delay is hundreds of msec but it's bit extreme).